controllers/etcdcluster_controller.go

```diff
@@ -121,35 +149,196 @@ func (r *EtcdClusterReconciler) reconcile(
+		} else {
+			// We can't contact the cluster, and we don't *think* we need more peers. So it's unsafe to take any
+			// action.
 		}
 	} else {
+		// In this branch we *do* have communication with the cluster, and have retrieved a list of members. We treat
+		// this list as canonical, and therefore reconciliation occurs in two cycles:
+		//
+		// 1. We reconcile the membership list to match our expected number of replicas (`spec.replicas`), adding
+		//    members one at a time until reaching the correct number.
+		// 2. We reconcile the number of peer resources to match the membership list.
+		//
+		// This 'two-tier' process may seem odd, but is designed to mirror etcd's operations guide for adding and
+		// removing members. During a scale-up, etcd itself expects that new peers will be announced to it *first* using
+		// the membership API, and then the etcd process started on the new machine. During a scale down the same is
+		// true, where the membership API should be told that a member is to be removed before it is shut down.
 		log.V(2).Info("Cluster communication established")
+
+		// We reconcile the peer resources against the membership list before changing the membership list. This is the
+		// opposite way around to what we just said. But is important. It means that if you perform a scale-up to five
+		// from three, we'll add the fourth node to the membership list, then add the peer for the fourth node *before*
+		// we consider adding the fifth node to the membership list.
+
+		// Add EtcdPeer resources for members that do not have one.
+		for _, member := range *members {
+			// Use this instead of member.Name. If we've just added the peer for this member then the member might not
+			// have that field set, so instead figure it out from the peerURL.
+			expectedPeerName, err := peerNameForMember(member)
+			if err != nil {
+				log.Error(err, "Found member with no identifiable name, unable to reconcile it, skipping")
+				continue
+			}
+			memberHasPeer := false
+			for _, peer := range peers.Items {
+				if expectedPeerName == peer.Name {
+					memberHasPeer = true
+					break
+				}
+			}
+			if !memberHasPeer {
+				// We have found a member without a peer. We should add an EtcdPeer resource for this member
+				peerName, err := peerNameForMember(member)
+				if err != nil {
+					return ctrl.Result{}, nil, err
+				}
+				log.V(1).Info(
+					"Found member in etcd's API that has no EtcdPeer resource representation, adding one.",
+					"member-name", peerName)
+				peer := peerForCluster(cluster, peerName)
+				configureJoinExistingCluster(peer, cluster, *members)
+
+				if err := r.Create(ctx, peer); err != nil {
+					return ctrl.Result{}, nil, fmt.Errorf("failed to create EtcdPeer %s: %w", peer.Name, err)
+				}
+				return ctrl.Result{}, &reconcilerevent.PeerCreatedEvent{Object: cluster, PeerName: peer.Name}, nil
+			}
+		}
+
+		// Remove EtcdPeer resources which do not have members.
+		// TODO Implement scale-down
```

Comment:
nit: in general, can we reference GH issues in TODOs so future readers can track the progress
```suggestion
		// TODO(#35) Implement scale-down
```